### PR TITLE
Detect Docusaurus files changes and trigger Jekyll build along with the Docusaurus build

### PR DESCRIPTION
**Pull request by ArturT**
Detect Docusaurus files changes and trigger Jekyll build along with the Docusaurus build.

Thanks to that you can run in your terminal:

```
$ guard
```

and in a separate window:

```
$ bundle exec jekyll serve --watch --future
```

When Guard detects changes in Docusaurus files then it touches the `.jekyll_build_trigger` file on the disk. This is detected by the command  `bundle exec jekyll serve --watch --future` and triggers Jekyll build. After Jekyll build the Docusaurus plugin (`_plugins/docusaurus.rb`) is triggered so Docusaurus is also build.

### CODE REVIEW

**Comment by 3v0k4** [comment]
I won't use it because it's way faster to `npm start` rather than `npm run build`. But it's probably an easier-temporary route for people who were used to running `guard` on this repo.

I tested and it works. Thanks!

**Comment by ArturT** [comment]
At first, I couldn't figure out how to use it just with `npm start`. Now I get it.

I'm putting this comment as a reference (just in case I forget how to use it).

# Docusaurus 

I do this:

```
cd docusaurus
npm install
npm start
```

Visit http://localhost:3000/knapsack_pro-ruby/guide/

When I change the content of the `docusaurus/docs/knapsack_pro-ruby/guide/index.md` file then, the http://localhost:3000/knapsack_pro-ruby/guide/ page refreshes automatically. It's fast, indeed.


# Jekyll

When I use Jekyll:

```
bundle exec jekyll serve --watch --future
```

and visit http://localhost:4000/knapsack_pro-ruby/guide/ (note there is 4000 port in the URL).
When I change the content of the `docusaurus/docs/knapsack_pro-ruby/guide/index.md` file, then I still see the old content on the page. Docusaurus changes are not reflected automatically, even when `npm start` is running. I need to use `guard` so that the Jekyll build is triggered and the plugin for Docusaurus is triggered (to copy Docusaurus pages to Jekyll destination directory). It works then, but it's slower (as you pointed out).

# In a nutshell

* If I want to work only with Docusaurus, then it's better to use `npm start` because it's fast.

* If I want to work with both docs (Jekyll with API in RAML + Docusaurus) at the same time, I can use `guard` and `bundle exec jekyll serve --watch --future`.

**Comment by 3v0k4** [comment]
What I do when working on Docusaurus is:

## Development

```bash
cd docusaurus
npm install
npm start
```

## Sanity check

Before deploying, I check the production build with:

```bash
npm install --global http-server # needed once
cd /this/repo/root
bundle exec jekyll build && http-server _site
```

But it's probably not so different than doing `guard` and `bundle exec jekyll serve --watch --future` as you mentioned.
